paasta_tools/marathon_tools.py

```diff
@@ -536,9 +547,19 @@ def get_host_port(self):
         '''Map this port on the host to your container's port 8888. Default is 0, which means Marathon picks a port.'''
         return self.config_dict.get('host_port', 0)
 
+    def get_marathon_shard(self):
+        """Returns the configued shard of Marathon to use.
+        Defaults to 0 currently as a fail-safe"""
+        return self.config_dict.get('marathon_shard', 0)
```

Comment:
can we rename this ``preferred_marathon_shard`` ? are we going to block deploying this if their shard isn't available?

Reply:
I can do that. Yea I think we kinda have to block if the destination shard is not available, otherwise we would be leaving old tasks behind. Also I don't want to implement hinted handoff.